docs/transpile/transpiler-plugins.ipynb

```diff
@@ -0,0 +1,306 @@
+{
```

Comment:
<p>I tried generating a random unitary and passing it through the <code>init</code> stage, and nothing seemed to happen. Could we have an example of the synthesis from the <code>init</code> stage (but before the <code>translation</code> stage takes is to the native basis gates of the backend)?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Qiskit/documentation/pull/781/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Reply:
Ok, here is my understanding.
- Unitary synthesis occurs in `init`, `translation`, and `optimization`.
- In `init`, only 3-qubit gates are synthesized. Furthermore, instances of UnitaryGate always use the same method (quantum Shannon decomposition) independent of what is passed as the argument of `unitary_synthesis_method`. During the `init` stage, that argument only affects the synthesis of gates that are not UnitaryGates.

@mtreinish Can you please confirm if this is correct or not?

Reply:
Ok I don't think any of that is particularly surprising. I have referred the reader to the page on transpiler stages for more details on the stages.

Reply:
LGTM!